### ISSUE TITLE
[BUG] <title>单卡微调报错，这里是全部的信息

### ISSUE THREAD

**Issue by dalingzina** [QwenLM/Qwen#557]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [X] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 当前行为 | Current Behavior

 sh finetune/finetune_lora_single_gpu.sh
[2023-10-31 06:10:42,570] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)
usage: finetune.py [-h] [--model_name_or_path MODEL_NAME_OR_PATH] [--data_path DATA_PATH] [--eval_data_path EVAL_DATA_PATH] [--lazy_preprocess [LAZY_PREPROCESS]] --output_dir OUTPUT_DIR
                   [--overwrite_output_dir [OVERWRITE_OUTPUT_DIR]] [--do_train [DO_TRAIN]] [--do_eval [DO_EVAL]] [--do_predict [DO_PREDICT]] [--evaluation_strategy {no,steps,epoch}]
                   [--prediction_loss_only [PREDICTION_LOSS_ONLY]] [--per_device_train_batch_size PER_DEVICE_TRAIN_BATCH_SIZE] [--per_device_eval_batch_size PER_DEVICE_EVAL_BATCH_SIZE]
                   [--per_gpu_train_batch_size PER_GPU_TRAIN_BATCH_SIZE] [--per_gpu_eval_batch_size PER_GPU_EVAL_BATCH_SIZE] [--gradient_accumulation_steps GRADIENT_ACCUMULATION_STEPS]
                   [--eval_accumulation_steps EVAL_ACCUMULATION_STEPS] [--eval_delay EVAL_DELAY] [--learning_rate LEARNING_RATE] [--weight_decay WEIGHT_DECAY] [--adam_beta1 ADAM_BETA1] [--adam_beta2 ADAM_BETA2]
                   [--adam_epsilon ADAM_EPSILON] [--max_grad_norm MAX_GRAD_NORM] [--num_train_epochs NUM_TRAIN_EPOCHS] [--max_steps MAX_STEPS]
                   [--lr_scheduler_type {linear,cosine,cosine_with_restarts,polynomial,constant,constant_with_warmup,inverse_sqrt,reduce_lr_on_plateau}] [--warmup_ratio WARMUP_RATIO] [--warmup_steps WARMUP_STEPS]
                   [--log_level {detail,debug,info,warning,error,critical,passive}] [--log_level_replica {detail,debug,info,warning,error,critical,passive}] [--log_on_each_node [LOG_ON_EACH_NODE]] [--no_log_on_each_node]
                   [--logging_dir LOGGING_DIR] [--logging_strategy {no,steps,epoch}] [--logging_first_step [LOGGING_FIRST_STEP]] [--logging_steps LOGGING_STEPS] [--logging_nan_inf_filter [LOGGING_NAN_INF_FILTER]]
                   [--no_logging_nan_inf_filter] [--save_strategy {no,steps,epoch}] [--save_steps SAVE_STEPS] [--save_total_limit SAVE_TOTAL_LIMIT] [--save_safetensors [SAVE_SAFETENSORS]]
                   [--save_on_each_node [SAVE_ON_EACH_NODE]] [--no_cuda [NO_CUDA]] [--use_cpu [USE_CPU]] [--use_mps_device [USE_MPS_DEVICE]] [--seed SEED] [--data_seed DATA_SEED] [--jit_mode_eval [JIT_MODE_EVAL]]
                   [--use_ipex [USE_IPEX]] [--bf16 [BF16]] [--fp16 [FP16]] [--fp16_opt_level FP16_OPT_LEVEL] [--half_precision_backend {auto,cuda_amp,apex,cpu_amp}] [--bf16_full_eval [BF16_FULL_EVAL]]
                   [--fp16_full_eval [FP16_FULL_EVAL]] [--tf32 TF32] [--local_rank LOCAL_RANK] [--ddp_backend {nccl,gloo,mpi,ccl,hccl}] [--tpu_num_cores TPU_NUM_CORES] [--tpu_metrics_debug [TPU_METRICS_DEBUG]]
                   [--debug DEBUG [DEBUG ...]] [--dataloader_drop_last [DATALOADER_DROP_LAST]] [--eval_steps EVAL_STEPS] [--dataloader_num_workers DATALOADER_NUM_WORKERS] [--past_index PAST_INDEX] [--run_name RUN_NAME]
                   [--disable_tqdm DISABLE_TQDM] [--remove_unused_columns [REMOVE_UNUSED_COLUMNS]] [--no_remove_unused_columns] [--label_names LABEL_NAMES [LABEL_NAMES ...]] [--load_best_model_at_end [LOAD_BEST_MODEL_AT_END]]
                   [--metric_for_best_model METRIC_FOR_BEST_MODEL] [--greater_is_better GREATER_IS_BETTER] [--ignore_data_skip [IGNORE_DATA_SKIP]] [--sharded_ddp SHARDED_DDP] [--fsdp FSDP]
                   [--fsdp_min_num_params FSDP_MIN_NUM_PARAMS] [--fsdp_config FSDP_CONFIG] [--fsdp_transformer_layer_cls_to_wrap FSDP_TRANSFORMER_LAYER_CLS_TO_WRAP] [--deepspeed DEEPSPEED]
                   [--label_smoothing_factor LABEL_SMOOTHING_FACTOR] [--optim OPTIM] [--optim_args OPTIM_ARGS] [--adafactor [ADAFACTOR]] [--group_by_length [GROUP_BY_LENGTH]] [--length_column_name LENGTH_COLUMN_NAME]
                   [--report_to REPORT_TO [REPORT_TO ...]] [--ddp_find_unused_parameters DDP_FIND_UNUSED_PARAMETERS] [--ddp_bucket_cap_mb DDP_BUCKET_CAP_MB] [--ddp_broadcast_buffers DDP_BROADCAST_BUFFERS]
                   [--dataloader_pin_memory [DATALOADER_PIN_MEMORY]] [--no_dataloader_pin_memory] [--skip_memory_metrics [SKIP_MEMORY_METRICS]] [--no_skip_memory_metrics]
                   [--use_legacy_prediction_loop [USE_LEGACY_PREDICTION_LOOP]] [--push_to_hub [PUSH_TO_HUB]] [--resume_from_checkpoint RESUME_FROM_CHECKPOINT] [--hub_model_id HUB_MODEL_ID]
                   [--hub_strategy {end,every_save,checkpoint,all_checkpoints}] [--hub_token HUB_TOKEN] [--hub_private_repo [HUB_PRIVATE_REPO]] [--hub_always_push [HUB_ALWAYS_PUSH]]
                   [--gradient_checkpointing [GRADIENT_CHECKPOINTING]] [--include_inputs_for_metrics [INCLUDE_INPUTS_FOR_METRICS]] [--fp16_backend {auto,cuda_amp,apex,cpu_amp}] [--push_to_hub_model_id PUSH_TO_HUB_MODEL_ID]
                   [--push_to_hub_organization PUSH_TO_HUB_ORGANIZATION] [--push_to_hub_token PUSH_TO_HUB_TOKEN] [--mp_parameters MP_PARAMETERS] [--auto_find_batch_size [AUTO_FIND_BATCH_SIZE]]
                   [--full_determinism [FULL_DETERMINISM]] [--torchdynamo TORCHDYNAMO] [--ray_scope RAY_SCOPE] [--ddp_timeout DDP_TIMEOUT] [--torch_compile [TORCH_COMPILE]] [--torch_compile_backend TORCH_COMPILE_BACKEND]
                   [--torch_compile_mode TORCH_COMPILE_MODE] [--dispatch_batches DISPATCH_BATCHES] [--include_tokens_per_second [INCLUDE_TOKENS_PER_SECOND]] [--cache_dir CACHE_DIR] [--model_max_length MODEL_MAX_LENGTH]
                   [--use_lora [USE_LORA]] [--lora_r LORA_R] [--lora_alpha LORA_ALPHA] [--lora_dropout LORA_DROPOUT] [--lora_target_modules LORA_TARGET_MODULES [LORA_TARGET_MODULES ...]] [--lora_weight_path LORA_WEIGHT_PATH]
                   [--lora_bias LORA_BIAS] [--q_lora [Q_LORA]]
finetune.py: error: the following arguments are required: --output_dir
finetune/finetune_lora_single_gpu.sh: 11: --model_name_or_path: not found
finetune/finetune_lora_single_gpu.sh: 12: --data_path: not found
finetune/finetune_lora_single_gpu.sh: 13: --bf16: not found
finetune/finetune_lora_single_gpu.sh: 14: --output_dir: not found
finetune/finetune_lora_single_gpu.sh: 15: --num_train_epochs: not found
finetune/finetune_lora_single_gpu.sh: 16: --per_device_train_batch_size: not found
finetune/finetune_lora_single_gpu.sh: 17: --per_device_eval_batch_size: not found
finetune/finetune_lora_single_gpu.sh: 18: --gradient_accumulation_steps: not found
finetune/finetune_lora_single_gpu.sh: 19: --evaluation_strategy: not found
finetune/finetune_lora_single_gpu.sh: 20: --save_strategy: not found
finetune/finetune_lora_single_gpu.sh: 21: --save_steps: not found
finetune/finetune_lora_single_gpu.sh: 22: --save_total_limit: not found
finetune/finetune_lora_single_gpu.sh: 23: --learning_rate: not found
finetune/finetune_lora_single_gpu.sh: 24: --weight_decay: not found
finetune/finetune_lora_single_gpu.sh: 25: --adam_beta2: not found
finetune/finetune_lora_single_gpu.sh: 26: --warmup_ratio: not found
finetune/finetune_lora_single_gpu.sh: 27: --lr_scheduler_type: not found
finetune/finetune_lora_single_gpu.sh: 28: --logging_steps: not found
finetune/finetune_lora_single_gpu.sh: 29: --report_to: not found
finetune/finetune_lora_single_gpu.sh: 30: --model_max_length: not found
finetune/finetune_lora_single_gpu.sh: 31: --lazy_preprocess: not found
finetune/finetune_lora_single_gpu.sh: 32: --gradient_checkpointing: not found
: not foundnetune_lora_single_gpu.sh: 33: --use_lora
: not foundnetune_lora_single_gpu.sh: 34:

### 运行环境 | Environment

```Markdown
- OS: Ubuntu 20.04
- Python:3.10
- Transformers:4.34.0
- PyTorch:2.0.0
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`):11.2
```

**Comment by jklj077** [comment]
您有修改`finetune/finetune_lora_single_gpu.sh`这个文件吗？根据报错信息，可能是https://github.com/QwenLM/Qwen/blob/c00209f932c38c45ac885a4db7c2429885d93f5c/finetune/finetune_lora_single_gpu.sh#L14 前一行的末尾的`\`没了。

**Comment by dalingzina** [comment]
这是我的finetune/finetune_lora_single_gpu.sh文件，

export CUDA_DEVICE_MAX_CONNECTIONS=1
DIR=`pwd`
MODEL="/data1/yanlingling/LLM/qwen14b/qwen/Qwen-14B-Chat-Int4" # Set the path if you do not want to load from huggingface directly
# ATTENTION: specify the path to your training data, which should be a json file consisting of a list of conversations.
# See the section for finetuning in README for more information.
DATA=...
OUTPUT_PATH=...
export CUDA_VISIBLE_DEVICES=0
python finetune.py \
  --model_name_or_path $MODEL \
  --data_path $DATA \
  --bf16 True \
  --output_dir "/finetune_Qwen/output_model" \
  --num_train_epochs 5 \
  --per_device_train_batch_size 2 \
  --per_device_eval_batch_size 1 \
  --gradient_accumulation_steps 8 \
  --evaluation_strategy "no" \
  --save_strategy "steps" \
  --save_steps 1000 \
  --save_total_limit 10 \
  --learning_rate 3e-4 \
  --weight_decay 0.1 \
  --adam_beta2 0.95 \
  --warmup_ratio 0.01 \
  --lr_scheduler_type "cosine" \
  --logging_steps 1 \
  --report_to "none" \
  --model_max_length 512 \
  --lazy_preprocess True \
  --gradient_checkpointing \
  --use_lora

没有少 \ 的

**Comment by InaR-design** [comment]
same issue here

**Comment by jklj077** [comment]
@dalingzina 用提供的脚本我们未能复现该问题。

根据报错现象，我们猜测是您编辑文件时，原始的换行被修改了(`\n`被改成了`\r\n`)。

请执行下以下命令：
```bash
file finetune/finetune_lora_single_gpu.sh
```
如果返回内容中有`with CRLF line terminators`，您需要修复换行问题。

**Comment by InaR-design** [comment]
> @dalingzina 用提供的脚本我们未能复现该问题。
> 
> 根据报错现象，我们猜测是您编辑文件时，原始的换行被修改了(`\n`被改成了`\r\n`)。
> 
> 请执行下以下命令：
> 
> ```shell
> file finetune/finetune_lora_single_gpu.sh
> ```
> 
> 如果返回内容中有`with CRLF line terminators`，您需要修复换行问题。

谢谢，确实是这个问题 可以使用dos2unix命令转化sh文件
